FiniteStateMachine/src/demo/Gearbox.java

```diff
@@ -1,8 +1,8 @@
 package demo;
 
-import statemachine.StateContainer;
+import statemachine.StateMachine;
 
-final class Gearbox extends StateContainer<Gear> {
+final class Gearbox extends StateMachine<Gear> {
```

Comment:
It's a state machine in the sense that it has a current state, and can transition to a set of other states, where that set is defined by the current state.